olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -301,6 +293,9 @@ class OlpClient::OlpClientImpl {
       const ParametersType& query_params, const ParametersType& header_params,
       const RequestBodyType& post_body, const std::string& content_type) const;
 
+  void AddBearer(const std::string& path, const ParametersType& query_params,
+                 http::NetworkRequest& request) const;
```

Comment:
I would not try to merge apples with peaches.
First we have the general url creation from base_, path and query_params which will be used later on by the OlpClient in scope of OLPEDGE-1805 to merge similar requests. Then once the URL inkl. query parameters are assembled we proceed to add token or apiKey in the last step.